src/BSON/Javascript.c

```diff
@@ -46,10 +46,17 @@ PHONGO_API zend_class_entry *php_phongo_javascript_ce;
 
 zend_object_handlers php_phongo_handler_javascript;
 
-/* Initialize the object from a string and return whether it was successful. */
+/* Initialize the object and return whether it was successful. An exception will
+ * be thrown on error. */
 static bool php_phongo_javascript_init(php_phongo_javascript_t *intern, const char *code, phongo_zpp_char_len code_len, zval *scope TSRMLS_DC)
 {
 	if (scope && Z_TYPE_P(scope) != IS_OBJECT && Z_TYPE_P(scope) != IS_ARRAY && Z_TYPE_P(scope) != IS_NULL) {
+		phongo_throw_exception(PHONGO_ERROR_INVALID_ARGUMENT TSRMLS_CC, "Expected %s scope to be array or object, %s given", ZSTR_VAL(php_phongo_javascript_ce->name), zend_get_type_by_const(Z_TYPE_P(scope)));
```

Comment:
Note: this exception can only be thrown from `__set_state()` or `unserialize()`, since `__construct()` validates the type with zpp.